config/value.go

```diff
@@ -476,13 +497,19 @@ func (cv Value) valueStruct(key string, target interface{}) (interface{}, error)
 				val = fieldInfo.DefaultValue
 			}
 			if val != nil {
-				v3, err := convertValue(val, fieldValue.Type())
-				if err != nil {
+				ret, err := convertValue(val, fieldType)
+				// For Unmarshalled text, if convertValue fails, and not returns
+				// non nil value, we fail populateStruct call by returning the error
+				if ret != nil && err != nil {
```

Comment:
nit: might be more readable to have
if ret != nil {
  if err != nil {
     return nil, err
  } else {
     fieldValue.set
  }
} else {
 if err := convertValueFromStruct; err != nil {
   return nil, err
 }
}